### PR TITLE
chore(react-tree): restructure stories, add separate category for flat tree

### PR DESCRIPTION
**Pull request by bsunderhus**
## Previous Behavior

Tree stories were split into:

1. Tree
2. TreeItem
3. Layouts

## New Behavior

Splits stories into:

1. Tree
2. TreeItem
3. Layouts
4. flat Tree

This way, scenarios that are specific to a flat tree can be properly segregated from the rest

### CODE REVIEW

**Comment by fabricteam** [comment]
## 📊 Bundle size report

<sub>🤖 This report was generated against <a href='https://github.com/microsoft/fluentui/commit/27665739a1f38468c63ea1544a2e82ff357d25f3'>27665739a1f38468c63ea1544a2e82ff357d25f3</a></sub>
<div id="bundleSizeComment9423"/>

**Comment by fabricteam** [comment]
## [Perf Analysis (`@fluentui/react-components`)](https://github.com/microsoft/fluentui/wiki/Perf-Testing)
<p>No significant results to display.</p><details><summary>All results</summary><p>
  <table>
  <tr>
    <th>Scenario</th>
    <th>Render type</th>
    <th>
      <a href="https://github.com/microsoft/fluentui/wiki/Perf-Testing#why-are-results-listed-in-ticks-instead-of-time-units">Master Ticks</a>
    </th>
    <th>
      <a href="https://github.com/microsoft/fluentui/wiki/Perf-Testing#why-are-results-listed-in-ticks-instead-of-time-units">PR Ticks</a>
    </th>
    <th>Iterations</th>
    <th>Status</th>
  </tr><tr>
            <td>Avatar</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/27586/perf-test-react-components/Avatar-mount_base.html">733</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/27586/perf-test-react-components/Avatar-mount.html">729</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>Button</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/27586/perf-test-react-components/Button-mount_base.html">401</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/27586/perf-test-react-components/Button-mount.html">405</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>Field</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/27586/perf-test-react-components/Field-mount_base.html">1245</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/27586/perf-test-react-components/Field-mount.html">1240</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>FluentProvider</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/27586/perf-test-react-components/FluentProvider-mount_base.html">784</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/27586/perf-test-react-components/FluentProvider-mount.html">780</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>FluentProviderWithTheme</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/27586/perf-test-react-components/FluentProviderWithTheme-mount_base.html">108</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/27586/perf-test-react-components/FluentProviderWithTheme-mount.html">109</a></td>
            <td>10</td>
            <td></td>
           </tr>
<tr>
            <td>FluentProviderWithTheme</td>
            <td>virtual-rerender</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/27586/perf-test-react-components/FluentProviderWithTheme-virtual-rerender_base.html">97</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/27586/perf-test-react-components/FluentProviderWithTheme-virtual-rerender.html">90</a></td>
            <td>10</td>
            <td></td>
           </tr>
<tr>
            <td>FluentProviderWithTheme</td>
            <td>virtual-rerender-with-unmount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/27586/perf-test-react-components/FluentProviderWithTheme-virtual-rerender-with-unmount_base.html">104</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/27586/perf-test-react-components/FluentProviderWithTheme-virtual-rerender-with-unmount.html">98</a></td>
            <td>10</td>
            <td></td>
           </tr>
<tr>
            <td>InfoButton</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/27586/perf-test-react-components/InfoButton-mount_base.html">21</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/27586/perf-test-react-components/InfoButton-mount.html">19</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>MakeStyles</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/27586/perf-test-react-components/MakeStyles-mount_base.html">1198</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/27586/perf-test-react-components/MakeStyles-mount.html">1187</a></td>
            <td>50000</td>
            <td></td>
           </tr>
<tr>
            <td>Persona</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/27586/perf-test-react-components/Persona-mount_base.html">1963</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/27586/perf-test-react-components/Persona-mount.html">1902</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>SpinButton</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/27586/perf-test-react-components/SpinButton-mount_base.html">1481</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/27586/perf-test-react-components/SpinButton-mount.html">1482</a></td>
            <td>5000</td>
            <td></td>
           </tr></table></p></details>

<div id="perfComment9425"/>

File: packages/react-components/react-tree/.storybook/tsconfig.json [hunk 1]
```diff
@@ -6,5 +6,10 @@
     "checkJs": true,
     "types": ["static-assets", "environment", "storybook__addons"]
   },
-  "include": ["../stories/**/*.stories.ts", "../stories/**/*.stories.tsx", "*.js"]
+  "include": [
+    "../stories/**/*.stories.ts",
```
Comment (Hotell) [inline packages/react-components/react-tree/.storybook/tsconfig.json:10]: looks like some editor/IDE magic gone wrong. 

pls undo this change

**Comment by Hotell** [comment]
Can you provide some more context for this PR ?

This doesn't say much
image


ty

**Comment by bsunderhus** [comment]
> Can you provide some more context for this PR ?
> 
> This doesn't say much image
> 
> ty

Yup, sorry there. Forgot this PR was not a draft anymore 😅

**Comment by ling1726** [comment]
I would put the Virtualization example first in the flat tree section since, it's one of the only two use cases you would really need a flat tree for

**Review by ling1726 — APPROVED** [review]
